src/app/modules/epidemiologia/services/ficha-epidemiologia.service.ts

```diff
@@ -25,4 +29,58 @@ export class FormsEpidemiologiaService extends ResourceBaseHttp {
             return seccionBuscada;
         }
     }
+
+    getConceptosCovidConfirmado(ficha) {
+        const conceptos = [];
+
+        const seccionClasificacion = this.getField(ficha, this.SECCION_CLASIFICACION_FINAL);
+
+        seccionClasificacion.fields.forEach(field => {
+            const key = Object.keys(field)[0];
+            switch (key) {
+                case 'segundaclasificacion':
+                    if (field.segundaclasificacion.id === 'confirmado') {
+                        conceptos.push(this.elementoRupService.getConceptoCovidConfirmadoNexo());
+                    }
+                    break;
+                case 'pcr':
+                    if (field.pcr.id === 'confirmado') {
+                        conceptos.push(this.elementoRupService.getConceptoConfirmadoPcr());
+                    };
+                    break;
+                case 'antigeno':
+                    if (field.antigeno.id === 'confirmado') {
+                        conceptos.push(this.elementoRupService.getConceptoConfirmadoTestRapido());
+                    } else if (field.antigeno.id === 'muestra') {
+                        conceptos.push(this.elementoRupService.getConceptoDescartadoTestRapido());
+                    }
+                    break;
+                case 'lamp':
+                    if (field.lamp.id === 'confirmado') {
+                        conceptos.push(this.elementoRupService.getConceptoEnfermedadCovid());
+                    };
+                    break;
+            }
+        });
+        return conceptos;
+    }
+
+    getConceptosCovidSospechoso(ficha) {
```

Comment:
No me parece buena idea agregar esta función. Fijate que ya estamos duplicando código para recuperar el antigeno negativo, y en caso de agregar o modificar conceptos, ya empezariamos a necesitar tocar código en dos lugares distintos... En su lugar usaría `getConceptosCovidConfirmado` (renombrada a `getConceptosCovid`), e incluiría ahi los conceptos de "caso sospechoso". Además habría que remover la condición de "confirmado" que usamos  para llamarla, y hacer esa evaluación dentro de la funcion (si clasificacion sospechosa,  entonces, `conceptos.push(this.elementoRupService.getConceptosCovidSospechoso())`....). 
Por otro lado, estaría bueno validar el orden... (si sospechoso va antes o despues de antigeno negativo).
Y últimamente: en los casos de c19 positivos con antígeno negativo, estaríamos regitrando dos prestaciones (1era caso sospechoso, y 2da de caso confirmado); esta bien registrar en dos prestaciones de seguimiento diferentes? o deberíamos registrar todo en la misma? 🤔

Reply:
Si totalmente de acuerdo, queda mejor renombrando la función `getConceptosCovidConfirmado` y hacer todo ahí.
Hice las modificaciones para usarla y comprobar ahí también la clasificación final.